### PR TITLE
Make the usage of XHarness via Helix SDK parallel-safe 

### LINKED ISSUE #5919: AddXHarnessCli in Helix SDK is not parallel-safe
dotnet/runtime makes parallel, parameterized submissions to Helix, mostly in GC stress testing different modes. This is a quick and effective way to submit the same job with different runtime parameters. For example, this block https://github.com/dotnet/runtime/blob/master/eng/pipelines/libraries/run-test-job.yml#L176-L184 causes the Helix submission to happen 7 times in parallel with different scenarios. This parallel launching is defined here: https://github.com/dotnet/runtime/blob/master/src/libraries/sendtohelix.proj#L23-L77

This does not work for cases where `<IncludeXHarnessCli>true</IncludeXHarnessCli>` is specified.

The `AddXHarnessCLI` target works by executing `dotnet tool install --tool-path /Users/directhex/Projects/runtime/artifacts/tmp/Release/Microsoft.DotNet.XHarness.CLI --version 1.0.0-prerelease.20403.2 --add-source https://dnceng.pkgs.visualstudio.com/public/_packaging/dotnet-eng/nuget/v3/index.json Microsoft.DotNet.XHarness.CLI` then specifying the tool's install directory as a Helix correlation payload - but fails when more than one Scenario is defined, because the `dotnet tool install` calls are not parallelisable.

Compare single-scenario:

```
  Done building Helix work items for scenario WasmTestOnBrowser. Work item count: 208
  You can invoke the tool using the following command: xharness
  Tool 'microsoft.dotnet.xharness.cli' (version '1.0.0-prerelease.20403.2') was successfully installed.
  Uploading payloads for Job on Ubuntu.1804.Amd64.Open...
```

to parallel-scenario:

```
  Done building Helix work items for scenario WasmTestOnBrowser. Work item count: 208
  Tool 'microsoft.dotnet.xharness.cli' (version '1.0.0-prerelease.20403.2') is already installed.
  Tool 'microsoft.dotnet.xharness.cli' failed to install. This failure may have been caused by:
  * You are attempting to install a preview release and did not use the --version option to specify the version.
  * A package by this name was found, but it was not a .NET tool.
  * The required NuGet feed cannot be accessed, perhaps because of an Internet connection problem.
  * You mistyped the name of the tool.
  For more reasons, including package naming enforcement, visit https://aka.ms/failure-installing-tool
/Users/directhex/Projects/runtime/.packages/microsoft.dotnet.helix.sdk/5.0.0-beta.20374.1/tools/xharness-runner/XHarnessRunner.targets(31,5): error MSB3073: The command "dotnet tool install --tool-path /Users/directhex/Projects/runtime/artifacts/tmp/Release/Microsoft.DotNet.XHarness.CLI --version 1.0.0-prerelease.20403.2 --add-source https://dnceng.pkgs.visualstudio.com/public/_packaging/dotnet-eng/nuget/v3/index.json Microsoft.DotNet.XHarness.CLI" exited with code 1. [/Users/directhex/Projects/runtime/src/libraries/sendtohelixhelp.proj]
##vso[task.logissue type=error;sourcepath=/Users/directhex/Projects/runtime/.packages/microsoft.dotnet.helix.sdk/5.0.0-beta.20374.1/tools/xharness-runner/XHarnessRunner.targets;linenumber=31;columnnumber=5;code=MSB3073;](NETCORE_ENGINEERING_TELEMETRY=Helix) The command "dotnet tool install --tool-path /Users/directhex/Projects/runtime/artifacts/tmp/Release/Microsoft.DotNet.XHarness.CLI --version 1.0.0-prerelease.20403.2 --add-source https://dnceng.pkgs.visualstudio.com/public/_packaging/dotnet-eng/nuget/v3/index.json Microsoft.DotNet.XHarness.CLI" exited with code 1.
  You can invoke the tool using the following command: xharness
  Tool 'microsoft.dotnet.xharness.cli' (version '1.0.0-prerelease.20403.2') was successfully installed.
  Uploading payloads for Job on Ubuntu.1804.Amd64.Open...
```

### PR DESCRIPTION
**Pull request by premun**
Fixes #5919

### CODE REVIEW

**Comment by premun** [comment]
cc @directhex

**Comment by directhex** [comment]
Well that's a simple approach

Is there a race condition? If thread A begins the installation of xharness into the staging directory, and thread B skips installation & marks it as a correlation payload before thread A finishes unpacking, could thread B submit work with an incomplete or invalid xharness install? I don't know the answer to that (I don't know the dotnet/nuget unpack flow, or if the move from "not unpacked" to "unpacked" is done in an atomic single step)

**Comment by premun** [comment]
Right.. so we need some kind of a lockfile then

**Comment by premun** [comment]
@directhex I cannot figure out how to synchronize it correctly other than ignore failures of the `dotnet tool install` command.. Otherwise, if two threads start downloading the NuGet at the same time, they will clash eventually.. No going around that

**Comment by premun** [comment]
@MattGal could you have a look at this one please? Thanks!

File: src/Microsoft.DotNet.Helix/Sdk/tools/xharness-runner/XHarnessRunner.targets [hunk 1]
```diff
@@ -25,20 +25,22 @@
...
+    <!-- It can happen that the SDK is invoked in parallel and then it can fail when re-using the install location, so we ignore the exit code -->
     <Exec Command="dotnet tool install --tool-path $(_XHarnessLocalToolPath) --version $(_XHarnessPackageVersion) --add-source $(XHarnessPackageSource) $(_XHarnessPackageName)"
-          WorkingDirectory="$(ArtifactsTmpDir)" />
+          WorkingDirectory="$(ArtifactsTmpDir)"
+          IgnoreExitCode="true" />
 
     <!-- There are files we don't need extracted inside of the extracted tool dir (such as the original .nupkg) that blow up the size 3x, so we remove them -->
     <ItemGroup>
-      <_XHarnessExtraToolFiles Include="$(ArtifactsTmpDir)\Microsoft.DotNet.XHarness.CLI\.store\microsoft.dotnet.xharness.cli\$(_XHarnessPackageVersion)\microsoft.dotnet.xharness.cli\$(_XHarnessPackageVersion)\*.*" />
+      <_XHarnessExtraToolFiles Include="$(_XHarnessCliPath)\*.*" />
```
Comment (MattGal) [inline src/Microsoft.DotNet.Helix/Sdk/tools/xharness-runner/XHarnessRunner.targets:38]: This globbing is non-recursive, is that intentional? (will ignore any extra tool files under sub-directories)

Comment (premun) [reply src/Microsoft.DotNet.Helix/Sdk/tools/xharness-runner/XHarnessRunner.targets:38]: Yes, we copy the rest in folders below. Should I add a comment?

Comment (MattGal) [reply src/Microsoft.DotNet.Helix/Sdk/tools/xharness-runner/XHarnessRunner.targets:38]: I don't think it matters enough to have a special comment, I was just checking my understanding

**Pull request merged**